src/service/resources-impl.js

```diff
@@ -387,15 +388,40 @@ export class Resources {
     this.resources_.push(resource);
 
     if (this.isRuntimeOn_) {
-      // Try to immediately build element, it may already be ready.
-      resource.build(this.forceBuild_);
-      this.schedulePass();
+      if (this.documentReady_) {
```

Comment:
Can we check if `element` is already parsed? That'll save pushing and splicing from `pendingBuildResources_` (and a lot of looping).